### PR TITLE
Update prometheus annotation example in docs of Autodiscover #16592

### PR DESCRIPTION
**Pull request by tgogos**
## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works

## Related  #issues

- Closes https://github.com/elastic/beats/issues/16592

### CODE REVIEW

File: heartbeat/docs/autodiscover-kubernetes-config.asciidoc [hunk 1]
```diff
@@ -9,12 +9,12 @@ heartbeat.autodiscover:
...
+              kubernetes.annotations.prometheus.io/scrape: "true"
           config:
             - type: http
               hosts: ["${data.host}:${data.port}"]
               schedule: "@every 1s"
               timeout: 1s
 -------------------------------------------------------------------------------------
 
-This configuration launches an `http` module for all containers of pods annotated with `prometheus.io.scrape=true`.
+This configuration launches an `http` module for all containers of pods annotated with `prometheus.io/scrape: "true"`.
```
Comment (ChrsMark) [inline heartbeat/docs/autodiscover-kubernetes-config.asciidoc]:
```suggestion
This configuration launches an `http` module for all containers of pods annotated with `prometheus.io/scrape=true`.
```

File: metricbeat/docs/autodiscover-kubernetes-config.asciidoc [hunk 2]
```diff
@@ -9,11 +9,11 @@ metricbeat.autodiscover:
...
-              kubernetes.annotations.prometheus.io.scrape: "true"
+              kubernetes.annotations.prometheus.io/scrape: "true"
           config:
             - module: prometheus
               metricsets: ["collector"]
               hosts: "${data.host}:${data.port}"
 -------------------------------------------------------------------------------------
 
-This configuration launches a `prometheus` module for all containers of pods annotated `prometheus.io.scrape=true`.
+This configuration launches a `prometheus` module for all containers of pods annotated `prometheus.io/scrape: "true"`.
```
Comment (ChrsMark) [inline metricbeat/docs/autodiscover-kubernetes-config.asciidoc]:
```suggestion
This configuration launches a `prometheus` module for all containers of pods annotated `prometheus.io/scrape=true`.
```

**Comment by ChrsMark** [comment]
jenkins, ok to test

**Comment by tgogos** [comment]
Thanks @ChrsMark for the guidance!

**Pull request merged**